### ISSUE TITLE
StackOverFlow Exception on Mac - fine on Windows

### ISSUE THREAD

**Issue by livarcocc** [dotnet/msbuild#2741]
_From @RickStrahl on November 22, 2017 3:53_

## Steps to reproduce
I just installed new 2.0.3 SDK on Mac and am trying to run a project that's working fine on Windows.

When I run any` dotnet` command I get:
![image](https://user-images.githubusercontent.com/1374013/33108732-ca17aed0-cee1-11e7-9340-e5b35fbc9215.png)

I'm using code from this repo:
https://github.com/RickStrahl/Westwind.Globalization

There are multiple .NET projects, the .NET Core one in question is the sample project in:
./src/NetCore/Westwind.Globalization.Sample.AspNetCore

and I'm running out of that folder.

This project has another project reference higher up the folder hierarchy so I'm wondering if that may be the problem. 

FWIW, I restore a couple other projects on the same machine and setup and using the same SDK and they work fine. Something is different but I can't figure out what 

## Expected  behavior
No stack overflow.

## Environment data
`dotnet --info` output:

Product Information:
 Version:            2.0.3
 Commit SHA-1 hash:  eb1d5ee318

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.13
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.0.3/

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.3
  Build    : a9190d4a75f4a982ae4b4fa8d1a24526566c69df

_Copied from original issue: dotnet/cli#8084_

**Comment by livarcocc** [comment]
Can you show what's the PWD that you are running this command from? Like, the full path? Also, any chance you have cyclic symbolic links somewhere in the directories used by the build? We have seen this in the past where we just keep following a cyclic symbolic link until we run out of memory.

I will try to repro this locally as well.

**Comment by livarcocc** [comment]
_From @RickStrahl on November 22, 2017 9:50_

Thanks for taking a look at this.

Your symbolic link suggestions pointed me into the right direction - there's an odd path that gets 'generated' when running on the Mac - as part of my application's code. This is a Windows app that's been moved to .NET Core and it appears there's something writing and errand file/path (backslash vs fwd slashes most likely).

Anyway this is the culprit:

![image](https://user-images.githubusercontent.com/1374013/33120576-93eaf72e-cf16-11e7-9dda-5248b290334f.png)

Now I need to figure out what's creating the file there. Ugh.

**Comment by livarcocc** [comment]
Makes sense.

I am going to go ahead and close this issue for now. If it turns out there is something else going on here, feel free to comment and re-activate.

Thanks for getting back to us on this.

**Comment by livarcocc** [comment]
_From @RickStrahl on November 22, 2017 21:48_

Yup, but there probably is a bug there anyway - no matter what we probably shouldn't see a StackOverflow exception.

In this case it's just a badly named folder (not a symlink). I think just having a sub-folder named `\` is the issue here.

**Comment by livarcocc** [comment]
You are right. We just checked and this happens inside the msbuild process.

I am going to go ahead and move it to msbuild's repo.

**Comment by rafiulgits** [comment]
![image](https://user-images.githubusercontent.com/27845214/97386845-f53b5c00-18fe-11eb-86fe-039d3c4b8f38.png)


Gettiing Stackoverflow with error code 134 while inserting a new entity in mssql-server. Currently using dotnet core 3.1.4, ef core 3.1.4 and mssql-server (default port on docker)

**Comment by danmosemsft** [comment]
If there is a way to use .NET 5.0, you might see more of a call stack for the stack overflow. (https://github.com/dotnet/runtime/pull/35418)

**Comment by YuliiaKovalova** [comment]
It looks like a stale issue.
Please open a new ticket if it remains a problem for you.
